exonum/src/node/state.rs

```diff
@@ -570,18 +578,27 @@ impl State {
 
     /// Adds the public key, address, and `Connect` message of a validator.
     pub fn add_peer(&mut self, pubkey: PublicKey, msg: Connect) -> bool {
-        self.connections.insert(msg.addr(), pubkey);
+        // This code is temporary solution until `Connect` message is removed,
+        // we should use ip address of the incoming message here but we use resolved public address instead.
```

Comment:
This may use an issue reference (ECR-XXXX) so that we do not forget to remove 'temporary' workarounds.